api/src/main/java/com/epam/pipeline/manager/execution/PipelineExecutor.java

```diff
@@ -160,14 +164,18 @@ private PodSpec getPodSpec(PipelineRun run, List<EnvVar> envVars, String secretN
                 .stream()
                 .anyMatch(env -> KubernetesConstants.CP_CAP_DIND_NATIVE.equals(env.getName()) &&
                         TRUE.equals(env.getValue()));
-        spec.setVolumes(getVolumes(isDockerInDockerEnabled));
+        boolean isSystemdEnabled = ListUtils.emptyIfNull(envVars)
```

Comment:
Maybe we should extract method `isParameterEnabled` and use it here and for lines 163-166?